### PR TITLE
add Docker and Kubernetes how-to instructions

### PR DESCRIPTION
**Pull request by lvnilesh**

### CODE REVIEW

File: docs/ReverseProxyToTor.md [hunk 1]
```diff
@@ -215,6 +215,127 @@ systemctl restart nginx
...
 Now, visiting `mydomain.com` should show your BTCPay Server instance.
 
+
+## Do all this in a Docker container
+
+Ready made docker image 
+
+  https://hub.docker.com/r/cloudgenius/socator
+
+Code behing the image
```
Comment (woutersamaey) [inline docs/ReverseProxyToTor.md:225]: "Behind" typo

File: docs/ReverseProxyToTor.md [hunk 2]
```diff
@@ -215,6 +215,127 @@ systemctl restart nginx
...
+
+  https://github.com/beacloudgenius/socator
+
+```
+### SocaTor = SOCAT + TOR
+Based on https://github.com/Arno0x/Docker-Socator
+
+It uses socat to listen on a given TCP port (5000 in this example) and to redirect incoming traffic to a tor hidden service specified through environment variables. Itacts as a relay between the standard web and a hidden service on the tor network. You can optionally restrict the IP addresses that are allowed to connect to this service by specifying an `ALLOWED_RANGE` environment variable and using CIDR notation.
```
Comment (woutersamaey) [inline docs/ReverseProxyToTor.md:233]: Forgot space "it acts"

File: docs/ReverseProxyToTor.md [hunk 3]
```diff
@@ -215,6 +215,127 @@ systemctl restart nginx
...
+        -e "TOR_SITE_PORT=80" \
+        --name socator \
+        cloudgenius/socato
+
+Now http://localhost:5000 should show you the tor hidden service you specified in the above command.
+
+```
+
+
+## Use that Docker container in a Kubernetes Cluster using these manifests
```
Comment (woutersamaey) [inline docs/ReverseProxyToTor.md:274]: Does BTCPay Server run inside Kubernetes? Didn't know we support that?

File: docs/ReverseProxyToTor.md (hunk 1 above)
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:225]:
```suggestion
Code behind the image
```

File: docs/ReverseProxyToTor.md (hunk 2 above)
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:233]:
```suggestion
It uses socat to listen on a given TCP port (5000 in this example) and to redirect incoming traffic to a Tor hidden service specified through environment variables. It acts as a relay between the standard web and a hidden service on the Tor network. You can optionally restrict the IP addresses that are allowed to connect to this service by specifying an `ALLOWED_RANGE` environment variable and using CIDR notation.
```

File: docs/ReverseProxyToTor.md (hunk 3 above)
Comment (dennisreimann) [reply docs/ReverseProxyToTor.md:274]: Hehe, me neither :)

@lvnilesh Can you use the code fence notation for the code samples as we do in the rest of the docs?

**Review by dennisreimann — CHANGES_REQUESTED** [review]

File: docs/ReverseProxyToTor.md (hunk 3 above)
Comment (lvnilesh) [reply docs/ReverseProxyToTor.md:274]: I find it best for privacy and security to run a full node, NBxplorer, and BTCpayserver all behind TOR and completely in my physical possession and full control.

However, I still want to expose the payment endpoint to the public clearnet without asking BTC payers to go through TOR. 

Enter socator container running in kubernetes.

In my example, I am running socator container in kubernetes in the cloud and everything else runs in my control and supervision.

I will examine the code fence notation next and update the PR.

File: docs/ReverseProxyToTor.md (hunk 1 above)
Comment (lvnilesh) [reply docs/ReverseProxyToTor.md]: Thanks.

**Review by lvnilesh — COMMENTED** [review]
Thanks.

File: docs/ReverseProxyToTor.md (hunk 1 above)
Comment (lvnilesh) [reply docs/ReverseProxyToTor.md]: Thanks for catching.

File: docs/ReverseProxyToTor.md (hunk 2 above)
Comment (lvnilesh) [reply docs/ReverseProxyToTor.md]: Thanks for catching.

File: docs/ReverseProxyToTor.md (hunk 3 above)
Comment (lvnilesh) [reply docs/ReverseProxyToTor.md:282]: code fence notation updated to match the rest.

**Comment by britttttk** [comment]
@lvnilesh preview your file to see formatting issues

File: docs/ReverseProxyToTor.md [hunk 4]
```diff
@@ -215,6 +215,135 @@ systemctl restart nginx
...
+Based on https://github.com/Arno0x/Docker-Socator
+
+It uses socat to listen on a given TCP port (5000 in this example) and to redirect incoming traffic to a Tor hidden service specified through environment variables. It acts as a relay between the standard web and a hidden service on the Tor network. You can optionally restrict the IP addresses that are allowed to connect to this service by specifying an `ALLOWED_RANGE` environment variable and using CIDR notation.
+
+### Usage
+
+Break free from cloud services providers limitations, secure and protect your bitcoin full node, connect that with a BTC Pay server, all behind TOR.
+Selectively expose the BTC Pay Server payment gateway and API to clearnet using socat+tor running on the Internet.
+
+--------------
```
Comment (britttttk) [inline docs/ReverseProxyToTor.md:240]: remove

File: docs/ReverseProxyToTor.md [hunk 5]
```diff
@@ -215,6 +215,135 @@ systemctl restart nginx
...
+
+```sh
+docker build -t cloudgenius/socator .
+```
+
+##### push
+
+```sh
+docker push cloudgenius/socator
+``
```
Comment (britttttk) [inline docs/ReverseProxyToTor.md:252]: missing ` to close formatting

File: docs/ReverseProxyToTor.md [hunk 6]
```diff
@@ -215,6 +215,135 @@ systemctl restart nginx
...
+
+Now http://localhost:5000 should show you the tor hidden service you specified in the above command.
+
+
+
+
+## Use that Docker container in a Kubernetes Cluster using these manifests
+
+```yaml
+---
```
Comment (britttttk) [inline docs/ReverseProxyToTor.md:285]: remove dividers and make each one separate yaml code block

File: docs/ReverseProxyToTor.md [hunk 7]
```diff
@@ -215,6 +215,135 @@ systemctl restart nginx
...
+
+
+### SocaTor = SOCAT + TOR
+Based on https://github.com/Arno0x/Docker-Socator
+
+It uses socat to listen on a given TCP port (5000 in this example) and to redirect incoming traffic to a Tor hidden service specified through environment variables. It acts as a relay between the standard web and a hidden service on the Tor network. You can optionally restrict the IP addresses that are allowed to connect to this service by specifying an `ALLOWED_RANGE` environment variable and using CIDR notation.
+
+### Usage
+
+Break free from cloud services providers limitations, secure and protect your bitcoin full node, connect that with a BTC Pay server, all behind TOR.
```
Comment (britttttk) [inline docs/ReverseProxyToTor.md:237]:
```suggestion
Break free from cloud services providers limitations, secure and protect your bitcoin full node, connect that with a BTCPay Server, all behind TOR.
```

File: docs/ReverseProxyToTor.md [hunk 8]
```diff
@@ -215,6 +215,135 @@ systemctl restart nginx
...
+Selectively expose the BTC Pay Server payment gateway and API to clearnet using socat+tor running on the Internet.
```
Comment (britttttk) [inline docs/ReverseProxyToTor.md:238]:
```suggestion
Selectively expose the BTCPay Server payment gateway and API to clearnet using socat+tor running on the Internet.
```

**Review by britttttk — CHANGES_REQUESTED** [review]

File: docs/ReverseProxyToTor.md (hunk 8 above)
Comment (lvnilesh) [reply docs/ReverseProxyToTor.md:243]: Thanks,

File: docs/ReverseProxyToTor.md (hunk 5 above)
Comment (lvnilesh) [reply docs/ReverseProxyToTor.md:257]: Thanks. Yikes. it appears I don't know how to write.

File: docs/ReverseProxyToTor.md (hunk 4 above)
Comment (lvnilesh) [reply docs/ReverseProxyToTor.md:245]: Sorry I did not understand this comment. Please help.

File: docs/ReverseProxyToTor.md (hunk 6 above)
Comment (lvnilesh) [reply docs/ReverseProxyToTor.md:295]: Manifests separated.

File: docs/ReverseProxyToTor.md (hunk 4 above)
Comment (britttttk) [reply docs/ReverseProxyToTor.md:245]: normally we're not using these dividers in the docs but overall it looks good :)

**Review by britttttk — APPROVED** [review]

File: docs/ReverseProxyToTor.md (hunk 6 above)
Comment (britttttk) [reply docs/ReverseProxyToTor.md:295]: just suggesting this to make it more beginner friendly :)

**Review by lvnilesh — COMMENTED** [review]
Thanks.

**Comment by lvnilesh** [comment]
This shows @dennisreimann requested changes but I can't see whats the request. Please describe. Thanks.

File: docs/ReverseProxyToTor.md [hunk 9]
```diff
@@ -215,6 +215,155 @@ systemctl restart nginx
...
+
+## Do all this in a Docker container
+
+Ready made docker image 
+
+  https://hub.docker.com/r/cloudgenius/socator
+
+Code behind the image
+
+  https://github.com/beacloudgenius/socator
```
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:227]:
```suggestion
Ready made [docker image](https://hub.docker.com/r/cloudgenius/socator) ([Code](https://github.com/beacloudgenius/socator))
```

File: docs/ReverseProxyToTor.md [hunk 10]
```diff
@@ -215,6 +215,155 @@ systemctl restart nginx
...
+
+
+### SocaTor = SOCAT + TOR
+Based on https://github.com/Arno0x/Docker-Socator
```
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:231]:
```suggestion
Based on [Docker-Socator](https://github.com/Arno0x/Docker-Socator)
```

File: docs/ReverseProxyToTor.md [hunk 11]
```diff
@@ -215,6 +215,155 @@ systemctl restart nginx
...
+
+Please note: 
+
+This container does not have any nginx component because Kubernetes provides for it. 
+
+
+### Usage
+
+Break free from cloud services providers limitations, secure and protect your bitcoin full node, connect that with a BTC Pay server, all behind TOR.
+Selectively expose the BTC Pay Server payment gateway and API to clearnet using socat+tor running on the Internet.
```
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:243]:
```suggestion
Selectively expose the BTCPay Server payment gateway and API to clearnet using socat+tor running on the Internet.
```

File: docs/ReverseProxyToTor.md [hunk 12]
```diff
@@ -215,6 +215,155 @@ systemctl restart nginx
...
+        -e "TOR_SITE_PORT=80" \
+        --name socator \
+        cloudgenius/socator
+```
+
+Now http://localhost:5000 should show you the tor hidden service you specified in the above command.
+
+
+
+
```
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:286]:
```suggestion

```

File: docs/ReverseProxyToTor.md [hunk 13]
```diff
@@ -215,6 +215,155 @@ systemctl restart nginx
...
+      labels:
+        role: socator
+    spec:
+      containers:
+      - image: cloudgenius/socator # code https://github.com/beacloudgenius/socator
+        imagePullPolicy: IfNotPresent
+        name: socator
+        env:
+          - name: TOR_SITE
+            value: "zqktlwiuavvvqqt4ybvgvi7tyo4hjl5xgfuvpdf6otjiycgwqbym2qad.onion" # BTCpayserver tor address => docker exec tor cat /var/lib/tor/app-btcpay-server/hostname
```
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:316]:
```suggestion
            value: "zqktlwiuavvvqqt4ybvgvi7tyo4hjl5xgfuvpdf6otjiycgwqbym2qad.onion" # BTCPay Server Tor address => docker exec tor cat /var/lib/tor/app-btcpay-server/hostname
```

File: docs/ReverseProxyToTor.md [hunk 14]
```diff
@@ -215,6 +215,155 @@ systemctl restart nginx
...
+  tls:
+    - hosts:
+        - btcpayserver.mydomain.com
+      secretName: socator-tls
+
+```
+
 ## Resources
 
 * [Nginx reverse proxy to .onion site in TOR network](https://itgala.xyz/nginx-reverse-proxy-to-onion-site-in-tor-network/)
```
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:369]:
```suggestion
* [nginx reverse proxy to .onion site in Tor network](https://itgala.xyz/nginx-reverse-proxy-to-onion-site-in-tor-network/)
```

**Review by dennisreimann — APPROVED** [review]

File: docs/ReverseProxyToTor.md [hunk 15]
```diff
@@ -215,9 +215,148 @@ systemctl restart nginx
...
+```sh
+docker run --rm -ti \
+        -p 5000:5000 \
+        -e "TOR_SITE=zqktlwiuavvvqqt4ybvgvi7tyo4hjl5xgfuvpdf6otjiycgwqbym2qad.onion" \
+        -e "TOR_SITE_PORT=80" \
+        --name socator \
+        cloudgenius/socator
+```
+
+Now http://localhost:5000 should show you the tor hidden service you specified in the above command.
```
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:276]:
```suggestion
Now http://localhost:5000 should show you the tor hidden service you specified in the above command.

```

File: docs/ReverseProxyToTor.md [hunk 16]
```diff
@@ -215,9 +215,148 @@ systemctl restart nginx
...
+## Use that Docker container in a Kubernetes Cluster using these manifests
+
+These manifest assumes a typical Kubernetes cluster that exposes internal services (like socator running internallly at port 5000) to the clearnet/public internet via Nginx Ingress https://github.com/kubernetes/ingress-nginx and provide automated Let's Encrypt TLS/SSL certificates via https://github.com/jetstack/cert-manager. 
+
+
```
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:281]:
```suggestion

```

**Comment by dennisreimann** [comment]
Just some more minor formatting stuff … thanks for the contribution! ACK :cake:

File: docs/ReverseProxyToTor.md [hunk 17]
```diff
@@ -215,9 +215,147 @@ systemctl restart nginx
 
 Now, visiting `mydomain.com` should show your BTCPay Server instance.
 
+
+## Do all this in a Docker container
+
+Ready made [docker image](https://hub.docker.com/r/cloudgenius/socator) ([Code](https://github.com/beacloudgenius/socator))
+
```
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:222]:
```suggestion
Ready made [Docker image](https://hub.docker.com/r/cloudgenius/socator) ([Code](https://github.com/beacloudgenius/socator))
```

File: docs/ReverseProxyToTor.md [hunk 18]
```diff
@@ -215,9 +215,147 @@ systemctl restart nginx
...
+### SocaTor = SOCAT + TOR
+Based on [Docker-Socator](https://github.com/Arno0x/Docker-Socator)
+
+It uses socat to listen on a given TCP port (5000 in this example) and to redirect incoming traffic to a Tor hidden service specified through environment variables. It acts as a relay between the standard web and a hidden service on the Tor network. You can optionally restrict the IP addresses that are allowed to connect to this service by specifying an `ALLOWED_RANGE` environment variable and using CIDR notation.
+
+Please note: 
+
+This container does not have any nginx component because Kubernetes provides for it. 
+
+
```
Comment (dennisreimann) [inline docs/ReverseProxyToTor.md:233]:
```suggestion

```

**Pull request merged**